webpay/libwebpay/TransbankSdkWebpay.php

```diff
@@ -10,21 +10,16 @@ class TransbankSdkWebpay {
     var $transaction;
 
     function __construct($config) {
+        $this->log = new LogHandler();
         if (isset($config)) {
             $environment = isset($config["MODO"]) ? $config["MODO"] : 'INTEGRACION';
             $configuration = Configuration::forTestingWebpayPlusNormal();
-            $this->log = new LogHandler();
-            $configuration = new Configuration();
-            $configuration->setEnvironment($environment);
-            $configuration->setCommerceCode($config["COMMERCE_CODE"]);
-            $configuration->setPrivateKey($config["PRIVATE_KEY"]);
-            $configuration->setPublicCert($config["PUBLIC_CERT"]);
-            if (trim($config["WEBPAY_CERT"]) != '') {
-                $this->log->logDebug($environment. ' - Usando certificado webpay definido por el usuario');
-                $configuration->setWebpayCert($config["WEBPAY_CERT"]);
-            } else {
-                $this->log->logDebug($environment . ' - Usando certificado webpay predeterminado');
-                $configuration->setWebpayCert(Webpay::defaultCert($environment));
+            $configuration->setWebpayCert(Webpay::defaultCert($environment));
```

Comment:
Sounds like this line should go inside the if block (because if we are on integration then `Configuration::forTestingWebpayPlusNormal();` would have set the correct certificate and this line would be redundant)

Reply:
I am not sure that Configuration::forTestingWebpayPlusNormal() is setting the integration certificate.

Reply:
I see it is setting one on https://github.com/TransbankDevelopers/transbank-sdk-php/blob/75db05aeb86140c08eaf1c15f46d6a30d4c31460/lib/webpay/Configuration.php#L140

(If it weren't, it would not really work as a one-line configuration to start using the SDK, which is the intention of those `forTesting*` helper methods)

Reply:
Also, even calling `setWebpayCert()` on the `if` block is something that shouldn't be required. If the environment is set to `"PRODUCTION"` the SDK should use the right webpay certificate automatically. 

It's also what I understand when reading the docs on https://transbankdevelopers.cl/documentacion/webpay?php#apuntar-a-produccion

Reply:
yes, I already applied the change :)

Reply:
Digging deeper I see that on the Java SDK we are doing that "automatic certificate configuration" on https://github.com/TransbankDevelopers/transbank-sdk-java/blob/a3c52327706d035d3241b58bec49a493d920a974/src/main/java/cl/transbank/webpay/Webpay.java#L139 (unless the SDK user has manually set a certificate). 

But on the PHP SDK I can't find any code that automatically picks the right webpay certificate based on the environment. That sounds like a bug that should be fixed. It's unreasonable to be asking PHP people to write `$configuration->setWebpayCert(Webpay::defaultCert($environment));` after they set the environment if that's something we could have done in the SDK itself and make things easier for people.

Reply:
@leosoto I found this https://github.com/TransbankDevelopers/transbank-sdk-php/blob/d78a320a6ced0d95aaaa7a5fd9ead1325ea70ca8/lib/webpay/Configuration.php#L140   https://github.com/TransbankDevelopers/transbank-sdk-php/blob/master/lib/webpay/Webpay.php. Is this what you meant?

Reply:
We talk about it and we will create a new card to do the change in the php sdk